source/DataHolder/AMDSDataHolder.cpp

```diff
@@ -74,6 +130,56 @@ AMDSDataTypeDefinitions::DataType AMDSFullDataHolder::dataType() const{
 	return oneFlatArray.dataType();
 }
 
+void AMDSFullDataHolder::cloneData(AMDSDataHolder *sourceDataHolder)
+{
+	AMDSFullDataHolder *sourceFullDataHolder = qobject_cast<AMDSFullDataHolder *>(sourceDataHolder);
+	if (sourceFullDataHolder) {
+		axes_.clear();
+		if (lightWeightDataHolder_)
+			lightWeightDataHolder_->deleteLater();
+
+		AMDSDataHolder *newDataHolder = AMDSDataHolderSupport::instantiateDataHolderFromInstance(sourceFullDataHolder->lightWeightDataHolder_);
+		lightWeightDataHolder_ = qobject_cast<AMDSLightWeightDataHolder *>(newDataHolder);
+		(*lightWeightDataHolder_) = (*sourceFullDataHolder->lightWeightDataHolder_);
```

Comment:
Is there a reason for exposing the control explicitly like this rather than using getters?